simpleocr/files.py

```diff
@@ -45,45 +32,152 @@ def write(self):
     """
-
-    def __init__(self, image_path):
-        if not os.path.exists(image_path):
-            raise IOError("Image file not found: " + image_path)
-        self.image_path = image_path
-        self.image = cv2.imread(self.image_path)
-        basepath = os.path.splitext(image_path)[0]
-        self.ground_path = try_extensions(GROUND_EXTENSIONS, basepath)
-        if self.ground_path:
-            self.ground = GroundFile(self.ground_path)
-            self.ground.read()
+    def __init__(self, array=None, path=None, debug=False):
+        # If array is actually a numpy array the truth value is ambiguous so it should be compared to None
+        if array is None:
+            # If array and path are both None, the object is created without any image data, so raise a ValueError
+            if not path:
+                raise ValueError("Image instance cannot be created without either an image array or an image path")
+            # Make sure the path is an absolute path to the image
+            path = self.get_absolute_path(path)
+            # Read the image and get an OpenCV compatible array
+            array = cv2.imread(path)
+        # Set all the attributes for the instance
+        self.image = array
+        self.path = path
+        self._debug = debug
+        # If a path is set, then the file is created with a file reference, so the existence of a ground file should
+        # be checked, and if it's available, the old ground data should be restored to memory
+        if self.path:
+            # Get the correct path for a ground file
+            self.ground_path = try_extensions(GROUND_EXTENSIONS, os.path.splitext(self.path)[0])
+            if os.path.exists(self.ground_path):
+                # Read the old ground data
+                self.ground = GroundFile(self.ground_path)
+                self.ground.read()
+            else:
+                # If the file is not available, set ground data to None
+                self.ground = None
         else:
-            self.ground_path = basepath + GROUND_EXTENSIONS_DEFAULT
+            # No file reference is available, so no ground data is available as the file has not yet been grounded
             self.ground = None
+            self.ground_path = None
+
+    @staticmethod
+    def from_file(path):
+        """
+        Create an image object with a path to an image. As the path is
+        put through the get_absolute_path function, the valid inputs are the same
+        as for the path passed to that function
+        :param path: image path, str
+        :return: Image object with file reference
+        """
+        path = Image.get_absolute_path(path)
+        array = cv2.imread(path)
+        return Image(array=array, path=path)
+
+    @staticmethod
+    def from_pil(pillow):
+        """ Create an Image object based off of a Pillow Image """
+        return Image(Image.get_array_from_pil(pillow))
+
+    @staticmethod
+    def get_absolute_path(image):
```

Comment:
Do you mean only taking an absolute path as a parameter in the instance creation? Because I had actually intended this whole function as the go-to function to get the whole path-finding out of the way.